runhouse/servers/cluster_servlet.py

```diff
@@ -258,46 +238,43 @@ async def aperiodic_cluster_checks(self):
                     break
 
                 logger.debug("Performing cluster checks")
-                status: ResourceStatusData = await self.astatus()
+                status, den_resp = await self.astatus(
+                    send_to_den=should_send_status_and_logs_to_den
+                )
+
                 if should_update_autostop:
                     logger.debug("Updating autostop")
                     await self._update_autostop(status)
 
                 if not should_send_status_and_logs_to_den:
                     break
 
-                logger.debug("Sending cluster status to Den")
-                cluster_rns_address = cluster_config.get("name")
-                cluster_uri = rns_client.format_rns_address(cluster_rns_address)
-                api_server_url = status.cluster_config.get(
-                    "api_server_url", rns_client.api_server_url
-                )
-
-                resp = await ClusterServlet.save_status_metrics_to_den(
-                    status=status,
-                    cluster_uri=cluster_uri,
-                    api_server_url=api_server_url,
-                )
-                status_code = resp.status_code
+                status_code = den_resp.status_code
 
                 if status_code == 404:
                     logger.info(
                         "Cluster has not yet been saved to Den, cannot update status or logs."
                     )
                 elif status_code != 200:
                     logger.error(
-                        f"{status_code}: Failed to send cluster status to Den: {resp.json()}"
+                        f"Failed to send cluster status to Den: {den_resp.json()}"
                     )
                 else:
+
                     logger.debug("Successfully sent cluster status to Den.")
+
                     prev_end_log_line = cluster_config.get("end_log_line", 0)
                     (
                         logs_resp_status_code,
                         new_start_log_line,
                         new_end_log_line,
                     ) = await self.send_cluster_logs_to_den(
-                        cluster_uri=cluster_uri,
-                        api_server_url=api_server_url,
+                        cluster_uri=rns_client.format_rns_address(
+                            self.cluster_config.get("name")
+                        ),
+                        api_server_url=self.cluster_config.get(
```

Comment:
why do we need to pass in `cluster_uri` and `api_server_url` here? doesnt the cluster servlet already have these values?

Reply:
yeah, agreed we could use the the cluster config we've previously loaded. 

I pass the `api_server_url` because we'll load it from the cluster config anyway, so we either load if from the config before passing it to `send_cluster_logs_to_den`, or we pass the config we loaded to `send_cluster_logs_to_den` and load the api_server_url there. Since we already loaded the cluster config in `aperiodic_cluster_checks` (which calls to `send_cluster_logs_to_den` I don't see a reason to load the cluster config again in `send_cluster_logs_to_den`). wdyt?